src/MALA.jl

```diff
@@ -0,0 +1,60 @@
+using .ForwardDiff: gradient!
+using .DiffResults: GradientResult, value, gradient
+
+struct MALA{D} <: MHSampler
+    proposal::D
+end
+
+
+# Create a RandomWalkProposal if we weren't given one already.
+MALA(d) = MALA(RandomWalkProposal(d))
+
+# If we were given a RandomWalkProposal, just use that instead.
+MALA(d::RandomWalkProposal) = MALA{typeof(d)}(d)
+
+
+struct GradientTransition{T<:Union{Vector, Real, NamedTuple}, L<:Real, G<:Union{Vector, Real, NamedTuple}} <: AbstractTransition
+    params :: T
+    lp :: L
+    gradient :: G
```

Comment:
Just some minor style comment :slightly_smiling_face: 
```suggestion
    params::T
    lp::L
    gradient::G
```